### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by Marcraphael12**
In this PR, we are working on the calculator's functionalities

- Implement the event handlers you need to use the math logic
- use state and props

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
+                {result}
+              </h3>
+            </li>
+            {buttons.map((btn) => (<li className={`pad ${addClass(btn)}`} key={btn}><button key={btn} onClick={this.eventhandler} type="button">{btn}</button></li>))}
+        </ul>
+      </div>
     );
   }
 }
-export default Component;
```
Comment (Emmyn5600) [inline src/components/calculator.js:41]: - [x] I had this while trying to run the project locally, to check if the project works perfectly. Please fix it.
user-3

 - [x] To implement this milestone, you are also supposed or required to use lifecycle methods.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi,

Good job so far! 👏🏼 @Marcraphael12 
 
## STATUS: CHANGES REQUESTED :recycle:


### Changes requested.

- Check the comment under the review.

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/components/calculator.js (hunk 1 above)
Comment (Marcraphael12) [reply src/components/calculator.js:41]: fixed

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @Marcraphael12, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [x] The Error mentioned in the previous review is still there. Kindly address the issue and fix it.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/App.css [hunk 2]
```diff
@@ -23,11 +23,23 @@ ul {
 .screen {
   width: 100%;
   border: 1px solid black;
+  height: auto;
+  display: flex;
+  align-items: center;
+  padding: 0 10px;
+  justify-content: flex-end;
 }
```
Comment (EricMbouwe) [inline src/App.css:31]: - [OPTIONAL] Make The Calculator.js component match the following given design:
Screenshot from 2021-11-03 14-05-23

**Review by EricMbouwe — APPROVED** [review]
### Hi @Marcraphael12, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.css (hunk 2 above)
Comment (Marcraphael12) [reply src/App.css:31]: i'll check that

**Comment by Temmarie** [comment]
Hello  @Marcraphael12     :wave:

_This is an additional review as a result of the second opinion request._

Since the issue has been resolved and the project has been approved I will go ahead and close this ticket. All the best in your next projects!!
Keep rocking and happy coding:rocket:!

**Pull request merged**